### PR TITLE
perf: replace useClick hook with onClick handler

### LINKED ISSUE #20177: [Task]: Refactor and revist the useClick hook
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

We have a delay of 200ms before we pass the click event to the handler. Removing this will give a very noticeable improvement in UX for the users.
https://github.com/appsmithorg/appsmith/blob/release/app/client/src/utils/hooks/useClick.tsx#L25

### PR DESCRIPTION
**Pull request by alex-golovanov**
## Description
Replaced useClick hook with onClick handler for improved performance.  More context is in the linked task below.

Fixes #20177 

## Automation

/ok-to-test tags="@tag.IDE"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10060525282>
> Commit: 0bb69f776f0f5333b58a1bc58f407cac356d9f43
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10060525282&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.IDE
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/ExplorerTests/API_Pane_spec.js
> <li>cypress/e2e/Regression/ClientSide/ExplorerTests/CopyQuery_RenameDatasource_spec.js
> <li>cypress/e2e/Regression/ClientSide/ExplorerTests/Pages_spec.js
> <li>cypress/e2e/Regression/ClientSide/ExplorerTests/Query_Datasource_spec.js
> <li>cypress/e2e/Regression/ClientSide/IDE/Canvas_Context_Selected_Widgets_spec.js</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Tue, 23 Jul 2024 14:37:44 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Added the `usehooks-ts` dependency for improved TypeScript hooks support.
	- Introduced an `onClick` prop to the `EntityName` component for enhanced interaction.

- **Bug Fixes**
	- Corrected casing of property names from `addButtonHelptext` to `addButtonHelpText` across multiple components for consistency.

- **Refactor**
	- Restructured properties in `EntityName` and `Entity` components for better readability and maintainability.
	- Simplified props handling and event management in the `Entity` component.

- **Style**
	- Improved code readability by correcting casing conventions in component props.

### CODE REVIEW

File: app/client/yarn.lock [hunk 1]
```diff
@@ -13395,6 +13395,7 @@ __metadata:
     typescript: 4.9.5
     unescape-js: ^1.1.4
     url-search-params-polyfill: ^8.0.0
+    usehooks-ts: ^3.1.0
```
Comment (KelvinOm) [inline app/client/yarn.lock:13398]: We should write to everyone to use these hooks.

File: app/client/src/pages/Editor/Explorer/Entity/index.tsx [hunk 2]
```diff
@@ -232,89 +227,119 @@ const SubItemWrapper = styled.div`
...
+      onToggle,
+      preRightIcon,
+      rightIcon,
+      runActionOnExpand,
+      searchKeyword,
+      showAddButton = false,
+      step,
+      updateEntityName,
+    } = props;
+
```
Comment (KelvinOm) [inline app/client/src/pages/Editor/Explorer/Entity/index.tsx:298]: 👍

**Review by KelvinOm — APPROVED** [review]

**Review by albinAppsmith — APPROVED** [review]
Ensure everything is working fine in EE.

**Comment by sagar-qa007** [comment]
I think, need to cypress case here. Please review UI changes before merging and if needed add cypress case.

**Comment by alex-golovanov** [comment]
> I think, need to cypress case here. Please review UI changes before merging and if needed add cypress case.

This is back in work, as I've overlooked some complexity in composition. It may not even see the light of day. Anyway, to answer your concern, functionality is not being changed here, so there is no need for an extra test case.

**Pull request closed without merging**